packages/next-auth/src/providers/facebook.ts

```diff
@@ -18,13 +18,13 @@ export default function Facebook<
       url: "https://graph.facebook.com/me",
       // https://developers.facebook.com/docs/graph-api/reference/user/#fields
       params: { fields: "id,name,email,picture" },
-      async request({ tokens, client, provider }) {
```

Comment:
Unfortunately, the `request` method was added (see #1846) to support providers that do not follow the spec properly... It's a shame that Facebook doesn't do it. (Note, it might have changed since, but I haven't checked in a while). If we can get rid of it, I'm all for though. But there might be other providers still using it.

We should discuss if we want to start discouraging (read: **drop support**/`request`) providers that do not implement the spec 100% correctly...

What I absolutely want to avoid is to go back to handling provider-specific stuff in our core. (eg `if (provider.id === "facebook")`, etc.)